### ISSUE TITLE
Time Picker: High contrast borders are not fully visible in hover and focus states 

### ISSUE THREAD

**Issue by mpriour** [Esri/calcite-components#6118]
### Summary

The time-picker component's buttons and inputs are not fully drawn when hovering or keyboard focusing in Windows high contrast accessibility mode.

### Actual Behavior

See the image below, it demonstrates both hover and focus outline styles missing parts in high contrast mode.

<img width="260" alt="Screen Shot 2022-12-19 at 5 34 26 PM" src="https://user-images.githubusercontent.com/142636/208550246-cd8f9324-7c52-48de-9ca7-e1bc17db139f.png">

### Expected Behavior

All the correct hover and focus outlines should be fully visible in high contrast mode.

### Reproduction Sample

https://esri.github.io/calcite-components/?path=/story/components-controls-time-input-time-picker--simple

### Reproduction Steps

Open the reproduction sample (https://esri.github.io/calcite-components/?path=/story/components-controls-time-input-time-picker--simple) in Chrome or Edge.

On Mac
1. Open Dev Tools.
2. Click on Elements tab.
3. Next to the console sub tab, there should be a "Rendering" tab. (if not you will need to enable it)
4. On the Rendering tab choose Forced-colors: active from the "Emulate forced colors..." selector.
5. Observe the issue as you mouse around or use keyboard to change focus.

On Windows
1. Turn on High Contrast mode
2. Observe the issue as you mouse around or use keyboard to change focus. 

### Reproduction Version

latest

### Relevant Info

Windows in High Contrast mode

Mac in Chrome / Edge using the `forced-colors: active` rendering mode

### Esri team

ArcGIS Online

**Comment by geospatialem** [comment]
Verified in [Storybook](https://esri.github.io/calcite-components/?path=/story/components-controls-time-input-time-picker--simple).
